### PR TITLE
[ao] Added DetectorQConfigMapping class framework

### PR DESCRIPTION
**Pull request by vspenubarthi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83358
* __->__ #83454

Summary: This adds the framework for a DetectorQConfigMapping class that
stores the QConfigMapping generated by a specific detector and also
allows the merging of two of these DetectorQConfigMappings to merge
suggestions from different detectors since the logic interacts in
different ways.

This is meant to be used in the ModelReport class to be able to get
suggestions from each of the detectors in the report and merge them
together to return a single merged mapping suggestion.

Test Plan: python test/test_quantization.py TestFxDetectorQConfigMapping

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

**Comment by vspenubarthi** [comment]
Created extraneously

File: torch/ao/quantization/fx/_model_report/detector.py [hunk 1]
```diff
@@ -1331,3 +1333,52 @@ def generate_detector_report(self, model: GraphModule) -> Tuple[str, Dict[str, A
             outlier_string += "There were no outliers found in the activations.\n"
 
         return (outlier_string, info_dict)
+
+
+# Mapping information
+# Adding class for returning QConfigMapping by Detector
+class DetectorQConfigMapping():
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_model_report/detector.py:1340]: maybe DetectorQConfigMappingGenerator?

File: torch/ao/quantization/fx/_model_report/detector.py [hunk 2]
```diff
@@ -1331,3 +1333,52 @@ def generate_detector_report(self, model: GraphModule) -> Tuple[str, Dict[str, A
...
+    needs to deal with one QConfigMapping that they can just use.
+
+    Args:
+        detector_classes ([DetectorBase]): The list of detector classes this
+            QConfigMapping was generated by.
+        mapping (QConfigMapping): The qconfig_mapping based on the suggestions
+            given by this class
+    """
+
+    def __init__(self, detector_classes: List[DetectorBase], mapping: QConfigMapping):
```
Comment (jerryzh168) [inline torch/ao/quantization/fx/_model_report/detector.py:1356]: why is `mapping` an argument? isn't it generated from detectors?

File: torch/ao/quantization/fx/_model_report/detector.py (hunk 1 above)
Comment (jerryzh168) [reply torch/ao/quantization/fx/_model_report/detector.py:1340]: also feel free to just implement it in this PR, no need to write the stub first

**Pull request closed without merging**